### PR TITLE
Update frames validation to CTD

### PR DESCRIPTION
**Pull request by kbeker**
Resolves https://github.com/golemfactory/concent/issues/645

### CODE REVIEW

File: concent_api/core/validation.py [hunk 1]
```diff
@@ -324,6 +324,12 @@ def validate_compute_task_def(compute_task_def: message.tasks.ComputeTaskDef) ->
                 ErrorCode.MESSAGE_INVALID
             )
 
+    if not extra_data.get('frames'):
```
Comment (rwrzesien) [inline concent_api/core/validation.py]: Change to `'frames' not in extra_data`.

File: concent_api/core/validation.py [hunk 2]
```diff
@@ -324,6 +324,12 @@ def validate_compute_task_def(compute_task_def: message.tasks.ComputeTaskDef) ->
...
+        raise ConcentValidationError(
+            "'extra_data' is missing in ComputeTaskDef",
```
Comment (rwrzesien) [inline concent_api/core/validation.py]: Please change the message, you have checked something different above.

File: concent_api/core/validation.py [hunk 3]
```diff
@@ -324,6 +324,12 @@ def validate_compute_task_def(compute_task_def: message.tasks.ComputeTaskDef) ->
                 ErrorCode.MESSAGE_INVALID
             )
 
+    if not extra_data.get('frames'):
+        raise ConcentValidationError(
```
Comment (Jakub89) [inline concent_api/core/validation.py]: unit test please :)

File: concent_api/core/validation.py (hunk 1 above)
Comment (kbeker) [reply concent_api/core/validation.py]: Done

File: concent_api/core/validation.py (hunk 2 above)
Comment (kbeker) [reply concent_api/core/validation.py]: Done

File: concent_api/core/validation.py [hunk 4]
```diff
@@ -324,9 +324,9 @@ def validate_compute_task_def(compute_task_def: message.tasks.ComputeTaskDef) ->
                 ErrorCode.MESSAGE_INVALID
             )
 
-    if not extra_data.get('frames'):
+    if not 'frames' in extra_data:
```
Comment (rwrzesien) [inline concent_api/core/validation.py]: `'frames' not in extra_data` , pylint will complain about it.

File: concent_api/core/validation.py [hunk 5]
```diff
@@ -324,9 +324,9 @@ def validate_compute_task_def(compute_task_def: message.tasks.ComputeTaskDef) ->
...
         raise ConcentValidationError(
-            "'extra_data' is missing in ComputeTaskDef",
+            "'frames' is missing in ComputeTaskDef",
```
Comment (rwrzesien) [inline concent_api/core/validation.py]: -> `'frames' is missing in ComputeTaskDef['extra_data']`

**Comment by kbeker** [comment]
I have found that this was already resolved earlier and error from Sentry showed up before it was merged. Resolved in: https://github.com/golemfactory/concent/pull/608.
Closing

**Pull request closed without merging**